src/network/core.rs

```diff
@@ -232,6 +232,210 @@ impl Core {
+
+            // Build up response information
+            let container_address: ipnet::IpNet =
+                match format!("{}/{}", static_ips[idx].to_string(), subnet_mask_cidr).parse() {
+                    Ok(i) => i,
+                    Err(e) => {
+                        return Err(Error::new(std::io::ErrorKind::Other, e));
+                    }
+                };
+            // Add the IP to the address_vector
+            address_vector.push(container_address);
+            response_net_addresses.push(types::NetAddress {
+                gateway: subnet.gateway,
+                ipnet: container_address,
+            });
+        }
+        debug!("Container macvlan name: {:?}", container_macvlan_name);
+        debug!("Master interface name: {:?}", master_ifname);
+        debug!("IP address for macvlan: {:?}", address_vector);
+
+        // create macvlan
+        let container_macvlan_mac = match Core::add_macvlan(
+            &master_ifname,
+            &container_macvlan_name,
+            macvlan_mode,
+            address_vector,
+            netns,
+        ) {
+            Ok(addr) => addr,
+            Err(err) => {
+                return Err(std::io::Error::new(
+                    std::io::ErrorKind::Other,
+                    format!("failed configure macvlan: {}", err),
+                ))
+            }
+        };
+        debug!("Container macvlan mac: {:?}", container_macvlan_mac);
+        let interface = types::NetInterface {
+            mac_address: container_macvlan_mac,
+            subnets: Option::from(response_net_addresses),
+        };
+        // Add interface to interfaces (part of StatusBlock)
+        interfaces.insert(container_macvlan_name, interface);
+        let _ = response.interfaces.insert(interfaces);
+        Ok(response)
+    }
+
+    pub fn add_macvlan(
+        master_ifname: &str,
+        container_macvlan: &str,
+        macvlan_mode: u32,
+        netns_ipaddr: Vec<ipnet::IpNet>,
+        netns: &str,
+    ) -> Result<String, std::io::Error> {
+        let _ = match core_utils::CoreUtils::configure_macvlan_async(
+            master_ifname,
+            container_macvlan,
+            macvlan_mode,
+            netns,
+        ) {
+            Ok(_) => (),
+            Err(err) => {
+                // it seems something went wrong
+                // we must not leave dangling interfaces
+                // otherwise cleanup would become mess
+                // try removing leaking interfaces from host
+                if let Err(er) = core_utils::CoreUtils::remove_interface(container_macvlan) {
+                    warn!("failed while cleaning up interfaces: {}", er);
+                }
+                return Err(std::io::Error::new(
+                    std::io::ErrorKind::Other,
+                    format!("failed while configuring macvlan {}:", err),
+                ));
+            }
+        };
+
+        match File::open(&netns) {
```

Comment:
This really needs to turn into a function or macro, I see it a lot in netavark code

Reply:
I agree there are multiple blocks I'll fix this in connection refactor PR.